frontend/components/Navbar.tsx

```diff
@@ -34,7 +42,17 @@ const globalMenuItems = [
   },
 ];
 
-function WrappedNavLink({ title, path, as, reactive }) {
+function hasSubmenuItems(
+  menuItem: GlobalMenuItem
+): menuItem is GlobalMenuItem & Required<Pick<GlobalMenuItem, "submenuItems">> {
+  return menuItem.submenuItems !== undefined;
+}
```

Comment:
We need a [type predicate](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates) here since TS cannot guarantee that the properties of an object have not been modified after checking their type.

Reply:
alrighty